docs/oras-auth-provider.md

```diff
@@ -78,39 +78,42 @@ Ratify pulls artifacts from a private Azure Container Registry using Workload Fe
 The official steps for setting up Workload Identity on AKS can be found [here](https://azure.github.io/azure-workload-identity/docs/quick-start.html).  
 
 1. Create ACR
-2. Create OIDC enabled AKS cluster by following steps [here](https://docs.microsoft.com/en-us/azure/aks/cluster-configuration#oidc-issuer-preview)
+2. Create OIDC enabled AKS cluster by following steps [here](https://learn.microsoft.com/en-us/azure/aks/cluster-configuration#oidc-issuer)
 3. Connect to cluster using kubectl by following steps [here](https://docs.microsoft.com/en-us/azure/aks/tutorial-kubernetes-deploy-cluster?tabs=azure-cli#connect-to-cluster-using-kubectl)
-4. Save the cluster's SERVICE_ACCOUNT_ISSUER (OIDC url): `az aks show --resource-group <resource_group> --name <cluster_name> --query "oidcIssuerProfile.issuerUrl" -otsv`
+4. Save the cluster's SERVICE_ACCOUNT_ISSUER (OIDC url): 
+```
+export SERVICE_ACCOUNT_ISSUER="$(az aks show --resource-group <resource_group> --name <cluster_name> --query "oidcIssuerProfile.issuerUrl" -otsv)"
+```
 5. Install Mutating Admission Webhook onto AKS cluster by following steps [here](https://azure.github.io/azure-workload-identity/docs/installation/mutating-admission-webhook.html)
 6. As the guide linked above shows, it's possible to use the AZ workload identity CLI or the regular az CLI to perform remaining setup. Following steps follow the AZ CLI.
-7. Create ACR AAD application: `az ad sp create-for-rbac --name "<APPLICATION_NAME>"`
+7. Create ACR AAD application: 
+```
+az ad sp create-for-rbac --name "<APPLICATION_NAME>"
+
+export APPLICATION_NAME=<APPLICATION_NAME>
+
+export APPLICATION_CLIENT_ID="$(az ad sp list --display-name "${APPLICATION_NAME}" --query '[0].appId' -otsv)"
+```
 8. On Portal or AZ CLI, enable acrpull role to the AAD application for the ACR resource
 ```
 // Sample AZ CLI command
-az role assignment create --assignee <AAD APPLICATION ID> --role acrpull --scope subscriptions/<SUBSCRIPTION NAME>/resourceGroups/<RESOURCE GROUP>/providers/Microsoft.ContainerRegistry/registries/<REGISTRY NAME>
+az role assignment create --assignee ${APPLICATION_CLIENT_ID} --role acrpull --scope subscriptions/<SUBSCRIPTION NAME>/resourceGroups/<RESOURCE GROUP>/providers/Microsoft.ContainerRegistry/registries/<REGISTRY NAME>
 ```
-9. Use kubectl to add service account to cluster: 
+9. Update `serviceAccount` attribute in ./charts/ratify/values.yaml file to add service account to cluster: 
 ```
-cat <<EOF | kubectl apply -f -
-apiVersion: v1
-kind: ServiceAccount
-metadata:
-  annotations:
-    azure.workload.identity/client-id: ${APPLICATION_CLIENT_ID}
-  labels:
-    azure.workload.identity/use: "true"
-  name: ${SERVICE_ACCOUNT_NAME}
-  namespace: ${SERVICE_ACCOUNT_NAMESPACE}
-EOF
+serviceAccount:
+  create: true
+  name: <Service Account Name>
 ```
-10. From Azure Cloud Shell: 
+10. From Azure Cloud Shell, `<Service Account Namespace>` is where you deploy Ratify:
```

Comment:
Out of curiosity, is behavior different if you do this in local bash shell? I'm trying to write a script that runs these steps and it seems like it's not strictly necessary for this to take place in the Azure Cloud Shell

Reply:
Oh that’s interesting. Would be great if it’s no longer required to be run in cloud shell. We added support for workload identity fairly early on which is when the guide required this step in cloud shell. It’s likely workload identity has evolved to not require it.

Reply:
Just confirming: it works without cloud shell?

Reply:
you're right, I just forgot to update this line. It works in my local shell.

Reply:
@binbin-li - should we update this as a minor fix?

Reply:
I'll make a one-liner fix.